syntax = "proto3";

package plotinus;

// 通用响应状态
message Status {
  int32 code = 1;
  string message = 2;
}

// 分页请求
message PaginationRequest {
  int32 page = 1;
  int32 per_page = 2;
}

// 分页响应
message PaginationResponse {
  int32 page = 1;
  int32 per_page = 2;
  int32 total_pages = 3;
  int32 total_items = 4;
  bool has_next = 5;
  bool has_prev = 6;
}

// 用户信息
message User {
  int32 id = 1;
  string username = 2;
}

// 空请求
message Empty {}
